apps/emqx_utils/src/emqx_utils_redact.erl

```diff
@@ -152,19 +152,24 @@ redact_v(_V) ->
     ?REDACT_VAL.
 
 deobfuscate(NewConf, OldConf) ->
+    deobfuscate(NewConf, OldConf, fun(_) -> false end).
+
+deobfuscate(NewConf, OldConf, IsSensitiveFun) ->
     maps:fold(
         fun(K, V, Acc) ->
             case maps:find(K, OldConf) of
                 error ->
-                    case is_redacted(K, V) of
+                    case is_redacted(K, V, IsSensitiveFun) of
                         %% don't put redacted value into new config
                         true -> Acc;
                         false -> Acc#{K => V}
                     end;
+                {ok, OldV} when is_map(V), is_map(OldV), ?IS_KEY_HEADERS(K) ->
```

Comment:
`redact` also handles when value is a tuple list, but i guess it's never the case in values needed to be deobfuscatted ?

Reply:
Good point.  It would be more symmetrical to treat those as well.  But, from what I can see, deobfuscation happens only at HTTP API boundaries (when receiving user input) and deals with raw configurations, so I think tuples are unexpected types in those situations.